jdaviz/configs/mosviz/plugins/viewers.py

```diff
@@ -88,14 +93,118 @@ def __init__(self, *args, **kwargs):
             "spectrum_2d_viewer_reference_name", "spectrum-2d-viewer"
         )
 
+        self.session.hub.subscribe(self, AddDataToViewerMessage,
+                                   handler=self._on_viewer_data_changed)
+        self.session.hub.subscribe(self, RemoveDataFromViewerMessage,
+                                   handler=self._on_viewer_data_changed)
+
+        self.scales['x'].observe(self._handle_x_axis_orientation, names=['min', 'max'])
+
+    @cached_property
+    def pixel_to_world_interp(self):
+        spectral_axis = self.state.reference_data.get_object().spectral_axis.value
+        pixels = range(len(spectral_axis))
+        return interp1d(pixels, spectral_axis)
```

Comment:
This makes some assumption on how pixel is mapped to spectral axis, no? Do we have to document this somewhere? Will this assumption break in some cases?

And when will this kick in? When the WCS in unavailable?

Reply:
This is now _always_ used for mapping values within the range of the data.  We might be able to use WCS methods directly instead, but this adds the benefit of caching the actual callable.  I don't think there are any extra assumptions here (please correct me if I'm wrong) and don't think the user cares about the internals of axes syncing to justify space in the docs, but I can add a docstring explaining the intent and that its meant to be used internally only for axes syncing.

Reply:
Is there no use cases with non-linear pixel-to-wavelength mapping? If not, then you can ignore...

Reply:
there is, see the examples in the PR description for how these are handled.  The axes limits within range are still linearly interpolated - I'm not sure we need anything more advanced for these cases (i.e., sub-pixel non-linear precision for axes limits).

Reply:
Okay, if the non-linear case is thought of and handled somewhere else, that is fine. Can you please add an in-line comment to explain this? Thanks!